### PR TITLE
Add authentication for routes

### LINKED ISSUE #140: Authentication for routes

### PR DESCRIPTION
**Pull request by Pr1sm0**

### CODE REVIEW

File: frontend/src/components/pages/profile/Profile.tsx [hunk 1]
```diff
@@ -11,11 +11,7 @@ import FriendsList from './components/friendsList/FriendsList';
 import Posts from './components/posts/Posts';
 import './profile.scss';
 
-interface MatchParams {
-  id: number & string;
-}
-
-interface IParamsProps extends RouteComponentProps<MatchParams> {}
+interface IParamsProps extends RouteComponentProps<any> {}
```
Comment (apastuhov) [inline frontend/src/components/pages/profile/Profile.tsx]: why any? use 
```
{match: {params: {id: string}}}
```

**Review by apastuhov — APPROVED** [review]
LGTM! There one small change, if there will be issues - it can be ignored

File: frontend/src/components/pages/profile/Profile.tsx (hunk 1 above)
Comment (oleksandrstarov) [inline frontend/src/components/pages/profile/Profile.tsx]: why any?

also this is nonsense type

interface MatchParams {
  id: number & string;
}

should be   id?:  string; (path params are only and always strings | undefined)

Comment (Pr1sm0) [reply frontend/src/components/pages/profile/Profile.tsx:14]: image

**Review by oleksandrstarov — APPROVED** [review]
LGTM

**Pull request merged**